### PR TITLE
Phase 1 for Digital River

### PR DESCRIPTION
**Pull request by aavila-vinneren**
Phase 1 release:

1. Added submenu for configuration keys and catalog sync logs
2. Change digital river key input type to password
3. Added call service to add initial setup, it creates MD table for catalog sync logs and required specs at a product level (ECCN, Country of origin)
4. Added configuration toggle for enable/disable full catalog sync
5. Catalog sync logs is showed in a pagination table and can filter by status(error/success), order by product id, sku, executed
6. Added service that adds specs product level (ECCN, Country of origin) and MD table
7. Add service to sync full catalog to DR catalog
8. Add trigger event to sync sku every time there is a sku change
9. Added in checkout custom js support for saving profile
10. Added in checkout custom js support for saving credit cards
11. Added in checkout custom js support for use stored credit cards
12. Added in checkout custom js support for compliance
13. Added in checkout custom js support for terms and conditions

Screenshots
Screen Shot 2021-09-05 at 6 14 27 AM

Screen Shot 2021-09-05 at 6 14 52 AM

Screen Shot 2021-09-05 at 6 15 05 AM

Screen Shot 2021-09-05 at 6 21 27 AM

### CODE REVIEW

File: react/CatalogLogs.tsx [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+
+                            trueKeys.forEach((key, index) => {
+                              trueKeysLabel += `${key}${
+                                index === trueKeys.length - 1 ? '' : ', '
+                              }`
+                            })
+
+                            return `${
+                              isAllTrue
+                                ? 'All'
```
Comment (polishq) [inline react/CatalogLogs.tsx:286]: Please replace with an intl message

File: react/CatalogLogs.tsx [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+                                : isAllFalse
+                                ? 'None'
```
Comment (polishq) [inline react/CatalogLogs.tsx:288]: Please replace with an intl message

File: react/CatalogLogs.tsx [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+                                : `${trueKeysLabel}`
+                            }`
+                          },
+                          verbs: [
+                            {
+                              label: 'Includes',
```
Comment (polishq) [inline react/CatalogLogs.tsx:294]: Please replace with an intl message

File: react/constants.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  return {
+    properties: {
+      error: {
+        title: 'Status',
+        width: 100,
+        // eslint-disable-next-line react/display-name
+        cellRenderer: ({ cellData }: { cellData: any }) => {
+          return (
+            <Tag bgColor={cellData ? '#ff4c4c' : '#8bc34a'} color="#FFFFFF">
+              <span className="nowrap">{cellData ? 'Error' : 'Success'}</span>
```
Comment (polishq) [inline react/constants.tsx:15]: Please replace 'Error' and 'Success' with intl messages; please do the same with each of the titles below

File: react/CatalogLogs.tsx [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+                  onClick={() =>
+                    getData(
+                      tableInfo.currentPage,
+                      dataSort,
+                      getStatusFilter(filterStatements)
+                    )
+                  }
+                  isLoading={syncLoading}
+                >
+                  Reload
```
Comment (polishq) [inline react/CatalogLogs.tsx:238]: Please replace with an intl message

File: react/CatalogLogs.tsx [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+                })}
+              >
+                <span className="mr4">
+                  <Button
+                    variation="primary"
+                    disabled={!enableSyncCatalog}
+                    onClick={() => handleSyncCatalog(showToast)}
+                    isLoading={syncLoading}
+                  >
+                    Sync Catalog
```
[the remark this answers is not in the document]
Comment (aavila-vinneren) [reply react/CatalogLogs.tsx]: Done

File: react/CatalogLogs.tsx [hunk 7]
```diff
@@ -0,0 +1,336 @@
...
+                    emptyStateLabel={tableInfo.emptyStateLabel}
+                    filters={{
+                      alwaysVisibleFilters: ['error'],
+                      statements: filterStatements,
+                      onChangeStatements: handleFiltersChange,
+                      clearAllFiltersButtonLabel: 'Clear Filters',
+                      collapseLeft: true,
+                      options: {
+                        error: {
+                          label: 'Status',
```
[the remark this answers is not in the document]
Comment (aavila-vinneren) [reply react/CatalogLogs.tsx]: Done

File: react/CatalogLogs.tsx [hunk 8]
```diff
@@ -0,0 +1,336 @@
...
+                          renderFilterLabel: (st: any) => {
+                            if (!st || !st.object) {
+                              return 'All'
```
[the remark this answers is not in the document]
Comment (aavila-vinneren) [reply react/CatalogLogs.tsx]: Done

File: react/CatalogLogs.tsx (hunk 1 above)
Comment (aavila-vinneren) [reply react/CatalogLogs.tsx]: Done

File: react/CatalogLogs.tsx (hunk 2 above)
Comment (aavila-vinneren) [reply react/CatalogLogs.tsx]: Done

File: react/CatalogLogs.tsx (hunk 3 above)
Comment (aavila-vinneren) [reply react/CatalogLogs.tsx]: Done

File: react/constants.tsx (hunk 4 above)
Comment (aavila-vinneren) [reply react/constants.tsx]: Done

File: react/CatalogLogs.tsx (hunk 5 above)
Comment (aavila-vinneren) [reply react/CatalogLogs.tsx]: Done

**Review by polishq — APPROVED** [review]

**Pull request merged**